libs/barista-components/filter-field/src/filter-field-multi-select/filter-field-multi-select-trigger.ts

```diff
@@ -0,0 +1,758 @@
+/**
+ * @license
+ * Copyright 2020 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { coerceBooleanProperty } from '@angular/cdk/coercion';
+import {
+  DOWN_ARROW,
+  ENTER,
+  ESCAPE,
+  SPACE,
+  TAB,
+  UP_ARROW,
+} from '@angular/cdk/keycodes';
+import {
+  Overlay,
+  OverlayConfig,
+  OverlayContainer,
+  OverlayRef,
+  PositionStrategy,
+  ViewportRuler,
+} from '@angular/cdk/overlay';
+import { Platform } from '@angular/cdk/platform';
+import { DOCUMENT } from '@angular/common';
+import {
+  ChangeDetectorRef,
+  Directive,
+  ElementRef,
+  Host,
+  Inject,
+  Input,
+  NgZone,
+  OnDestroy,
+  Optional,
+} from '@angular/core';
+import { ControlValueAccessor } from '@angular/forms';
+import {
+  calculateOptionHeight,
+  DtAutocompleteOrigin,
+  DtOptionConfiguration,
+  DT_AUTOCOMPLETE_VALUE_ACCESSOR,
+  DT_OPTION_CONFIG,
+} from '@dynatrace/barista-components/autocomplete';
+import {
+  DtFlexibleConnectedPositionStrategy,
+  DtOption,
+  DtOptionSelectionChange,
+  dtSetUiTestAttribute,
+  DtUiTestConfiguration,
+  DtViewportResizer,
+  DT_UI_TEST_CONFIG,
+  isDefined,
+  stringify,
+  _countGroupLabelsBeforeOption,
+  _getOptionScrollPosition,
+  _readKeyCode,
+} from '@dynatrace/barista-components/core';
+import { DtFormField } from '@dynatrace/barista-components/form-field';
+import {
+  defer,
+  EMPTY,
+  fromEvent,
+  merge,
+  Observable,
+  of as observableOf,
+  Subject,
+  Subscription,
+} from 'rxjs';
+import {
+  delay,
+  filter,
+  map,
+  startWith,
+  switchMap,
+  take,
+  takeUntil,
+  tap,
+} from 'rxjs/operators';
+import { DtFilterFieldMultiSelect } from './filter-field-multi-select';
+
+@Directive({
+  selector: `input[dtFilterFieldMultiSelect]`,
+  exportAs: 'dtFilterFieldMultiSelectTrigger',
+  host: {
+    class: 'dt-multiselect-trigger',
+    '[attr.autocomplete]': 'autocompleteAttribute',
+    '[attr.aria-activedescendant]': 'activeOption?.id',
+    '[attr.aria-expanded]': 'multiSelectDisabled ? null : panelOpen.toString()',
+    '[attr.aria-owns]':
+      '(multiSelectDisabled || !panelOpen) ? null : multiSelect?.id',
+    '(focusin)': '_handleFocus()',
+    '(blur)': '_handleBlur()',
+    '(input)': '_handleInput($event)',
+    '(keydown)': '_handleKeydown($event)',
+  },
+  providers: [DT_AUTOCOMPLETE_VALUE_ACCESSOR],
+})
+export class DtFilterFieldMultiSelectTrigger<T>
+  implements ControlValueAccessor, OnDestroy {
```

Comment:
Why does the Multiselect trigger need to implement the ControlValueAccessor? This trigger should function quite the same as the filterfield-range-trigger, but seems far more complicated. 
I don't quite understand why the trigger needs it's own autocomplete that it can disable, and why it needs a controlValueAccessor? 

Maybe you can explain this to me, because I think this can be done similarly to the filter-field-range-trigger.
Please ignore further comments in this file for now.

Reply:
After thinking about this a bit more, I do think I understand why you need access to the autocomplete. 

In order to filter down the choices that you present in the select, you need access to the input control. I do see that as problematic though, as we now control the autocomplete (input) from two directions. Once from the filterfield itself and once by the multiselect. 
I would suggest finding a way where the filterfield passes the current input value into multiselect, rather than accessing the whole autocomplete. 
Would that make sense?

Reply:
As fabian and tom suggested:
- Implementing ControlValueAccessor is not necessary.
- You share a lot of code with the range. Let's create a shared class that both extend from?